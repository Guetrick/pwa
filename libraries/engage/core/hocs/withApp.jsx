import React from 'react';
import AppContext from '../contexts/AppContext';

/**
 * Injects the AppContext properties into the desired component. It adds a new property "app".
 * @param {Function} WrappedComponent The react component to wrap.
 * @returns {JSX}
 */
export function withApp(WrappedComponent) {
  /**
   * The actual HOC.
   * @param {Object} props The component props.
   * @returns {JSX}
   */
  const WithApp = props => (
    <AppContext.Consumer>
      {appContext => (
        <WrappedComponent app={appContext} {...props} />
      )}
    </AppContext.Consumer>
  );

  const displayName = WrappedComponent.displayName || WrappedComponent.name || 'Component';
  WithApp.displayName = `WithApp(${displayName})`;

  return WithApp;
}
