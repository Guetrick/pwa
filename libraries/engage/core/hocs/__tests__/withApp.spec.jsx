import React from 'react';
import { mount } from 'enzyme';
import { withApp } from '../withApp';

const mockApp = {
  some: 'prop',
};

const mockContext = jest.fn().mockReturnValue(mockApp);

jest.mock('../../contexts/AppContext', () => ({
  Consumer: ({ children }) => children(mockContext()),
}));

// eslint-disable-next-line require-jsdoc
const MockComponent = () => null;

describe('engage > core > hocs > withApp', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should inject a property called "app" into the component', () => {
    const ComposedComponent = withApp(MockComponent);
    const wrapper = mount(<ComposedComponent someProp />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(MockComponent).props()).toEqual({
      someProp: true,
      app: {
        ...mockApp,
      },
    });
  });
});
