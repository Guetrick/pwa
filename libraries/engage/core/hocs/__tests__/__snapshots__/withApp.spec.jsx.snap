// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`engage > core > hocs > withApp should inject a property called "app" into the component 1`] = `
<WithApp(MockComponent)
  someProp={true}
>
  <Consumer>
    <MockComponent
      app={
        Object {
          "some": "prop",
        }
      }
      someProp={true}
    />
  </Consumer>
</WithApp(MockComponent)>
`;
